src/nupic/algorithms/Cells4.cpp

```diff
@@ -1628,46 +1667,55 @@ void Cells4::adaptSegment(const SegmentUpdate& update)
     // updateSynapses? It would be cleaner and avoid the need to keep the
     // synapses sorted.
 
-    // Accumulate list of synapses to decrement, increment, add, and remove
-    std::set<UInt> synapsesSet(update.begin(), update.end());
+    // Accumulate lists of synapses to decrement, increment, add, and remove
+    std::set<UInt> synapsesSet(update.begin(), update.end()); // srcCellIdx
```

Comment:
@scottpurdy, I explained above what the trailing comment means and that I borrowed from at least one existing use case in that same function. I understand that both the existing use case(s) and the new one(s) may be unclear to the reader/maintainer of the code as you have indicated.

My personal preference is to use better variable naming that is self-documenting as it makes it easier and less error-prone to follow the code without having to refer to those trailing comments. Unfortunately, that is not the case in this old body of code. Some of these variable names are used in other methods, such as `Segment::freeNSynapses`, as well as tests. It may not be practical to update them all as part of this task; that would arguably be out of scope of this task, especially considering current priorities and that this old code is no longer being actively developed.

I'd be happy to improve the existing and new trailing comments that describe the contents of the containers in this method. Do you have a suggestion? How about something like "src cell indexes" instead of srcCellIdx  and "synapse  indexes" instead of "syn indexes"?